dask/dataframe/core.py

```diff
@@ -4132,6 +4171,33 @@ def applymap(self, func, meta="__no_default__"):
     def round(self, decimals=0):
         return elemwise(M.round, self, decimals)
 
+    @derived_from(pd.DataFrame)
+    def mode(self, axis=0, numeric_only=False, dropna=True, split_every=False):
+        axis = self._validate_axis(axis)
+        if axis == 1:
+            return super(DataFrame, self).mode(
+                axis=axis, dropna=dropna, split_every=split_every
+            )
+        elif axis == 0:
+            from .multi import concat
+
+            mode_series_list = []
+            for col in self.columns:
+                mode_series_list.append(
+                    super(Series, self[col]).mode(
+                        dropna=dropna, split_every=split_every
+                    )
+                )
+
+            ddf = concat(mode_series_list, axis=1)
```

Comment:
@TomAugspurger Thank you for your response.  I wasn't aware of value_counts, but it promises to simplify things quite a bit.  I'm hitting an error here on `ddf = concat(mode_series_list, axis=1)`.  The error is below.
```python3  
File "/home/balast/PycharmProjects/dask/dask_repo/dask/dataframe/multi.py", line 865, in concat_and_check
    raise ValueError("Concatenated DataFrames of different lengths")
ValueError: Concatenated DataFrames of different lengths
```
There is no error when concatenating series of different lengths if the divisions are known, but the divisions aren't known here. I believe I can get around this using the delayed api similar to what I did in the last PR, where I add np.nans to each short array until they are all the same length, but before I do so, I was wondering if you have a suggestion to concatenate series of various lengths without using the delayed api.
but before I do I was wondering if you had a recommendation of a better way to do so?